ykrt/src/mt.rs

```diff
@@ -949,4 +957,41 @@ mod tests {
         mtt.control_point(loc, simple_interp_step, &mut ctx);
         assert_eq!(ctx.run, 2);
     }
+
+    #[test]
+    fn loop_interpreter() {
+        use std::{thread, time};
+        let mut mtt = MTBuilder::new().hot_threshold(2).init();
+
+        let loc = Location::new();
+
+        struct InterpCtx {
+            counter: u8,
+        }
+
+        #[interp_step]
+        fn simple_interp_step(ctx: &mut InterpCtx) -> bool {
+            let a = ctx.counter;
+            if ctx.counter < 100 {
+                ctx.counter = a + 1;
+            } else {
+                ctx.counter = a + 2;
+            }
+            if ctx.counter > 110 {
```

Comment:
This can just be `ctx.counter > 110` without the `if`/`return` dance.

Reply:
I'll have to test first that this still generates `return false` in SIR, which then triggers the rewrite to `LoopEnd` in TIR.

Reply:
This seems to work, although we'll soon replace the return type with an enum, in which case the if/else will come back. So we might as well leave it. Unless you feel strongly about it.

Reply:
OK.